src/main/java/io/cdap/plugin/http/common/BaseHttpConfig.java

```diff
@@ -0,0 +1,349 @@
+package io.cdap.plugin.http.common;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.plugin.http.common.http.AuthType;
+import io.cdap.plugin.http.common.http.OAuthUtil;
+import io.cdap.plugin.http.source.common.BaseHttpSourceConfig;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.apache.http.Header;
+
+/**
+ *
+ */
+public class BaseHttpConfig extends PluginConfig {
```

Comment:
1. This class should be an abstract class.
2. This class should be extending from the ReferencePluginConfig.